### ISSUE TITLE
Grid column header violates  accessibility rule - Name, Role, Value (Level A)

### ISSUE THREAD

**Issue by prabhukumarb08** [mui/mui-x#12515]
### Search keywords

Data Grid, Column header, ARIA role, ARIA attribute, Level A, column header description,  description, aria-label

### Latest version

- [X] I have tested the latest version

### Steps to reproduce

Link to live example: (https://mui.com/x/react-data-grid/column-header/#system-HeaderColumnsGrid.js)

Reproducible with any demo examples from [MUI Grid Docs](https://mui.com/x/react-data-grid/column-header/)

Steps:
1. Import and start using MUI GRID component.
2. Pass necessary basic props. Including columns
3. add description in column config (Not mandatory)
<img width="758" alt="image" src="https://github.com/mui/mui-x/assets/82257601/3f4511be-c0f2-461b-bc21-11fcd03982a8">

<img width="822" alt="image" src="https://github.com/mui/mui-x/assets/82257601/7d022ad4-3a27-4ffb-91d2-c07aaac8ae43">

<img width="745" alt="image" src="https://github.com/mui/mui-x/assets/82257601/13e2bc2c-5c34-4980-88c2-59265ea1d4a4">

### Current behavior

Nested Div element contains aria-lable without corresponding ARIA Role. Violation of accessibility rule - [Failure 59](https://www.w3.org/WAI/WCAG21/Techniques/failures/F59)  ![image](https://github.com/mui/mui-x/assets/82257601/f681428d-4af4-49c4-aa13-d1ff5b9a7131)

### Expected behavior

Elements use only ARIA attributes that are allowed for their corresponding ARIA role. Div should hold a appropriate role

### Context

Nested Div within column header holds aria-label without any corresponding role. Do not comply  Level A rule

Column header do not comply accessibility rule - [Name, Rule, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html)

No role, No aria-attributes [Failure 59](https://www.w3.org/WAI/WCAG21/Techniques/failures/F59)

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>
